app/src/main/java/uk/gov/onelogin/MainActivityViewModel.kt

```diff
@@ -62,6 +60,38 @@ class MainActivityViewModel @Inject constructor(
         }
     }
 
+    private fun handleTokens(tokens: TokenResponse) {
+        val jwksUrl = context.getString(
+            R.string.stsUrl,
+            context.getString(R.string.jwksEndpoint)
+        )
+
+        tokens.idToken?.let { idToken ->
+            verifyIdToken(idToken, jwksUrl) { verified ->
+                if (!verified) {
+                    _next.value = LoginRoutes.SIGN_IN_ERROR
+                } else {
+                    checkLocalAuthRoute(tokens)
+                }
+            }
+        } ?: checkLocalAuthRoute(tokens)
```

Comment:
if the idToken is missing we carry on with login?

Reply:
hmm, yeah alright. Maybe in future we will need to optionally disable the verify in lower environments like `dev`? or STS will be available in dev as well?